funflow/src/Control/FunFlow/External/Coordinator.hs

```diff
@@ -51,8 +51,7 @@ class Coordinator c where
   initialise :: MonadIO m => Config c -> m (Hook c)
 
   -- | Submit a task to the task queue.
-  --   If this task is already known to the system and not completed, then
```

Comment:
What's the motivation for this change? As I understand it, the executor should no longer store a 'completed' status, but I don't see why this would mean we'd want to restart 'in-progress' tasks?

Reply:
The flow interpreter should first query the store if the item is missing/pending/complete,
and it should only submit a task to the coordinator if the item is missing. If the store considers the item missing, but the coordinator has a running task for it then something must have gone wrong. In particular, that means that the build directory into which the task would store it's output no longer exists and that that task is going to fail once it tries to write its output. Ideally we would kill the running task before submitting the new task. But, I'm not sure that we can do that in the general case, e.g. when the task is being executed on a different machine.

In any case, if this happens, then the system is in an invalid state.
Maybe it's safer to throw an exception in this case, instead of quietly resubmitting?